llama_cpp/llama.py

```diff
@@ -814,60 +814,32 @@ def _create_completion(
                             break
 
                 token_end_position = 0
-                for token in remaining_tokens:
-                    token_end_position += len(self.detokenize([token]))
-                    # Check if stop sequence is in the token
-                    if token_end_position >= (
-                        remaining_length - first_stop_position - 1
-                    ):
+                idx = 0
+                while idx < len(remaining_tokens):
+                    tokens = []
+                    for i in range(0, len(remaining_tokens) - idx):
+                        tokens.append(remaining_tokens[idx+i])
+                        try:
+                            bs = self.detokenize(tokens)
+                            text = bs.decode('utf-8')
+                            break
+                        except Exception as _e:
+                            pass
+                    idx += len(tokens)
+                    token_end_position += len(bs)
+                    if token_end_position >= (remaining_length - first_stop_position - 1):
                         break
-                    logprobs_or_none: Optional[CompletionLogprobs] = None
-                    if logprobs is not None:
-                        token_str = self.detokenize([token]).decode(
```

Comment:
some CJK character contains several tokens,  while here it only decodes a single token, which causes this bug